iceoryx_binding_c/test/moduletests/test_client.cpp

```diff
@@ -139,15 +142,195 @@ TEST_F(iox_client_test, notInitializedOptionsAreUninitialized)
 #endif
 }
 
-TEST_F(iox_client_test, initializedOptionsAreInitialized)
+TEST_F(iox_client_test, InitializedOptionsAreInitialized)
 {
     ::testing::Test::RecordProperty("TEST_ID", "b512741e-9c1f-410f-a40b-68fec4a72bc5");
     iox_client_options_t initializedOptions;
     iox_client_options_init(&initializedOptions);
     EXPECT_TRUE(iox_client_options_is_initialized(&initializedOptions));
 }
 
-TEST_F(iox_client_test, initializedOptionsAreToCPPDefaults)
+TEST_F(iox_client_test, InitializedOptionsWithNullptrFails)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "3ae62644-5fb2-45cf-af99-b4daba43d044");
+    IOX_EXPECT_FATAL_FAILURE<iox::HoofsError>([&] { iox_client_options_init(nullptr); },
+                                              iox::HoofsError::EXPECTS_ENSURES_FAILED);
+}
+
+TEST_F(iox_client_test, CheckInitializedOptionsWithNullptrFails)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "9a3b7845-170f-4b7f-a0a4-f5b43d96059f");
+    iox_client_options_t initializedOptions;
+    iox_client_options_init(&initializedOptions);
+    IOX_EXPECT_FATAL_FAILURE<iox::HoofsError>([&] { iox_client_options_is_initialized(nullptr); },
+                                              iox::HoofsError::EXPECTS_ENSURES_FAILED);
+}
+
+TEST_F(iox_client_test, InitializingClientWithNullptrFails)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "ce04604b-ae5b-451f-842b-3c3d3f41ebb7");
+    iox_client_options_t options;
+    iox_client_options_init(&options);
+    options.responseQueueCapacity = 456;
+    strncpy(options.nodeName, "hypnotoad is all you need", IOX_CONFIG_NODE_NAME_SIZE);
+    options.connectOnCreate = false;
+    options.responseQueueFullPolicy = QueueFullPolicy_BLOCK_PRODUCER;
+    options.serverTooSlowPolicy = ConsumerTooSlowPolicy_WAIT_FOR_CONSUMER;
+
+    IOX_EXPECT_FATAL_FAILURE<iox::HoofsError>([&] { iox_client_init(nullptr, SERVICE, INSTANCE, EVENT, &options); },
+                                              iox::HoofsError::EXPECTS_ENSURES_FAILED);
+    IOX_EXPECT_FATAL_FAILURE<iox::HoofsError>([&] { iox_client_init(&sutStorage, nullptr, INSTANCE, EVENT, &options); },
+                                              iox::HoofsError::EXPECTS_ENSURES_FAILED);
+    IOX_EXPECT_FATAL_FAILURE<iox::HoofsError>([&] { iox_client_init(&sutStorage, SERVICE, nullptr, EVENT, &options); },
+                                              iox::HoofsError::EXPECTS_ENSURES_FAILED);
+    IOX_EXPECT_FATAL_FAILURE<iox::HoofsError>(
+        [&] { iox_client_init(&sutStorage, SERVICE, INSTANCE, nullptr, &options); },
+        iox::HoofsError::EXPECTS_ENSURES_FAILED);
+    // IOX_EXPECT_FATAL_FAILURE<iox::HoofsError>(
```

Comment:
```cpp
    iox::cxx::Expects(options == nullptr || (options != nullptr && iox_client_options_is_initialized(options)));
```
todo: check how to test this condition.

Reply:
I guess you can just create a `iox_client_options_t` on the stack. Call `iox_client_init` with that object and the exactly what you did in this commented out code :)